docs_src/microservices/configuration/ConfigurationAndRegistry.md

```diff
@@ -0,0 +1,186 @@
+
+## Configuration Provider
+
+You can supply and manage configuration in a centralized manner by utilizing the `-cp/--configProvider` flag when starting a service. If the flag is provided and points to an application such as [HashiCorp's Consul](https://www.consul.io/), the service will bootstrap its configuration into the provider, if it doesn't exist. If configuration does already exist, it will load the content from the given location applying any environment variables overrides of which the service is aware. Integration with the configuration provider is handled through the [go-mod-configuration](https://github.com/edgexfoundry/go-mod-configuration) module referenced by all services.
+
+## Registry Provider
+
+The registry refers to any platform you may use for service discovery. For the EdgeX Foundry reference implementation, the default provider for this responsibility is Consul. Integration with the registry is handled through the [go-mod-registry](https://github.com/edgexfoundry/go-mod-registry) module referenced by all services.
+
+### Introduction to Registry
+
+The objective of the registry is to enable micro services to find and to communicate with each other. When each micro service starts up, it registers itself with the registry, and the registry continues checking
+its availability periodically via a specified health check endpoint. When one micro service needs to connect to another one, it connects to the registry to retrieve the available host name and port number of the
+target micro service and then invokes the target micro service. The following figure shows the basic flow.
+
+![image](EdgeX_ConfigurationRegistry.png)
+
+Consul is the default registry implementation and provides native features for service registration, service discovery, and health checking. Please refer to the Consul official web site for more information:
+
+> <https://www.consul.io>
+
+Physically, the "registry" and "configuration" management services are combined and running on the same Consul server node.
+
+### Web User Interface
+
+A web user interface is also provided by Consul. Users can view the available service list and their health status through the web user interface. The web user interface is available at the /ui path on the same port as the HTTP API. By default this is <http://localhost:8500/ui>. For more detail, please see:
+
+> <https://www.consul.io/intro/getting-started/ui.html>
+
+### Running on Docker
+
+For ease of use to install and update, the micro services of EdgeX Foundry are published as Docker images onto Docker Hub, including Registry:
```

Comment:
Do we suggest user running Consul on non-secure mode for this section?

Reply:
I'll take a look.

Reply:
Reworked to point them to the compose files to run in Docker.